Bento/Adapters/AdapterStore.swift

```diff
@@ -0,0 +1,207 @@
+import FlexibleDiff
+
+/// The adapter store which carries the component tree, and also cached sizing information if enabled.
+///
+/// The adapter computes sizing information lazily upon request, but it adjusts spaces and performs cache invalidation
+/// every time a diff is applied.
+///
+/// - important: While the use-as-you-go Bento uses the adapter, the size caching capability should only be enabled in
+///              `SizeCachingTableView` and `SizeCachingCollectionView`, since there are specific messages that Bento
+///              need to intercept for size caching to function as expected.
+struct AdapterStore<SectionID: Hashable, ItemID: Hashable> {
```

Comment:
Why making it a struct? It seems to me that it behaves more like a ref type, no?

Reply:
It doesn't need identity. After all it is just operations upon a bunch of data.